raiden_libs/private_contract.py

```diff
@@ -11,13 +15,17 @@ def __init__(self, call):
 
     def transact(self, *args, **kwargs) -> bytes:
         """Creates a transaction, signs it with a provided private key and sends it. """
+        args = deepcopy(args)
         private_key = kwargs.pop('private_key', None)
         if len(args) == 0:
             args = ({},)
         # buildTransaction requires 'from' field to be set properly
         args[0]['from'] = private_key_to_address(private_key)
         gas_limit = args[0].pop('gas_limit', GAS_LIMIT_CONTRACT)
-        tx_data = self._call.buildTransaction(*args, **kwargs)
+        if isinstance(self._call, ContractConstructor):
```

Comment:
Please add some docs as to why we're distinguishing between how we call constructor from other methods?

Reply:
actually it works fine without the switch... :)